exonum/src/events/internal.rs

```diff
@@ -30,10 +36,32 @@ pub struct InternalPart {
 
 impl InternalPart {
     pub fn run(self, handle: Handle) -> Box<Future<Item = (), Error = io::Error>> {
+        // Default number of threads = number of cores.
+        let thread_pool = ThreadPoolBuilder::new().build();
+
+        // Buffer in a channel wouldn't do anything except clutter the memory.
+        let (pool_tx, pool_rx) = mpsc::channel::<Box<Transaction>>(0);
+        let internal_tx = self.internal_tx.clone();
+        thread_pool.spawn(futures::lazy(move || {
+            pool_rx.for_each(move |tx| {
+                if tx.verify() {
+                    internal_tx
+                        .clone()
+                        .send(InternalEvent::TxVerified(tx))
+                        .wait()
+                        .expect("Cannot send TxVerified event.");
```

Comment:
We doesn't use unwrap in futures, so can you throw error through futures chain?

Reply:
Sorry for my limited Futures knowledge, but what futures chain are you talking about? The motivation behind what I did there was to make `send` call synchronous, which is why I chose to use `expect` there. Besides, the future is not passed on anywhere.